### PR TITLE
Improve code quality based on linter hints

### PR DESCRIPTION
**Pull request by efd6**
Changes to remove magic constants, handle errors and clean up.

Some changes requested by opinionated editor/linter.

Makefile fixed to ensure that build directory exists before starting work.

Please take a look.

### CODE REVIEW

File: internal/files/compress.go [hunk 1]
```diff
@@ -36,7 +36,10 @@ func Zip(sourcePath, destinationFile string) error {
 	}
 	defer os.RemoveAll(tempDir)
 	workDir := filepath.Join(tempDir, folderNameFromFileName(destinationFile))
-	os.MkdirAll(workDir, 0755)
+	err = os.MkdirAll(workDir, 0755)
+	if err != nil {
+		return errors.Wrap(err, "can't prepare work directory")
```
Comment (mtojek) [inline internal/files/compress.go:41]: nit: `errors.Wrapf( ..., workDir)`?

File: Makefile [hunk 2]
```diff
@@ -26,15 +26,17 @@ update-readme:
 
 update: update-readme
 
-test-go:
+test-build-dir:
+	mkdir -p $(PWD)/build/test-results
```
Comment (mtojek) [inline Makefile:30]: That's funny as @stuartnelson3 found it in the same time [here](https://github.com/elastic/elastic-package/pull/565#discussion_r740779179) :)

I would ask you or @stuartnelson3 to extract changes in Makefile to a separate PR.

Comment (mtojek) [reply Makefile:30]: Ok, @stuartnelson3 did this already (thanks!). Would you mind adjusting your PR?

File: internal/files/compress.go (hunk 1 above)
Comment (efd6) [reply internal/files/compress.go:41]: That will be in the underlying `os.PathError`, but done. pkg/errors makes me sad.

File: Makefile (hunk 2 above)
Comment (efd6) [reply Makefile:30]: Done.

**Review by mtojek — APPROVED** [review]
Thanks! Let merge this on
